installer/scripts/omsadmin.sh

```diff
@@ -682,8 +682,11 @@ main()
     fi
 	
     # If we reach this point, onboarding was successful, we can remove the
-    # onboard conf to prevent accidentally re-onboarding
+    # onboard conf to prevent accidentally re-onboarding 
     [ "$ONBOARD_FROM_FILE" = "1" ] && rm "$FILE_ONBOARD" > /dev/null 2>&1 || true
+  
+    # Register omsagent as a service and start the agent
+    /opt/microsoft/omsagent/bin/service_control enable
```

Comment:
I don't think this is correct. We can onboard (up at line 669), and that can abort if there is a problem (or continue if not).

There's no conditional here. So even if we did NOT onboard, it seems like we would still enable the service, which is not correct here. Why would you not do this only if $ONBOARDING = 1 ?